src/mindlessgen/qm/xtb.py

```diff
@@ -71,13 +90,27 @@ def optimize(
             # read the optimized molecule
             optimized_molecule = molecule.copy()
             optimized_molecule.read_xyz_from_file(temp_path / "xtbopt.xyz")
-
+            if np.any(np.isin(molecule.ati, get_lanthanides())):
+                # Reset the UHF value to the original value before returning the optimized molecule.
+                optimized_molecule.uhf = uhf_original
             return optimized_molecule
 
     def singlepoint(self, molecule: Molecule, verbosity: int = 1) -> str:
         """
         Perform a single-point calculation using xtb.
         """
+        nel = 0
```

Comment:
Shouldn't we do this only in the case that lanthanides are used?

Reply:
Yes! i forgott to change this for the singlepoint aswell.